avro-builder/builder/src/test/java/com/linkedin/avroutil1/builder/SchemaBuilderTest.java

```diff
@@ -170,6 +171,40 @@ public void testImportableSchemasUsingOwnCodegen() throws Exception {
     Assert.assertEquals(javaFiles.size(), 2);
   }
 
+  public void testUnderscores(boolean useOwnCodegen) throws Exception {
+    File simpleProjectRoot = new File(locateTestProjectsRoot(), "test-underscores");
+    File inputFolder = new File(simpleProjectRoot, "input");
+    File outputFolder = new File(simpleProjectRoot, "output");
+    if (outputFolder.exists()) { //clear output
+      FileUtils.deleteDirectory(outputFolder);
+    }
+    //run the builder
+    SchemaBuilder.main(new String[] {
+        "--input", inputFolder.getAbsolutePath(),
+        "--output", outputFolder.getAbsolutePath(),
+        "--generator", useOwnCodegen ? CodeGenerator.AVRO_UTIL.name() : CodeGenerator.VANILLA.name()
+    });
+    //see output was generated
+    Optional<Path> javaFile = Files.find(outputFolder.toPath(), 5,
+        (path, basicFileAttributes) -> path.getFileName().toString().endsWith(".java")
+    ).collect(Collectors.toList()).stream().findFirst();
+    Assert.assertTrue(javaFile.isPresent());
+
+    String file = new String(Files.readAllBytes(javaFile.get()));
+
+    Assert.assertTrue(file.contains("getFieldWithUnderscores()"));
```

Comment:
probbaly not as part of this PR, but we need something like assertCompiles() to see that the resulting java file "works".

part of a bigger work-item